The test failure is from our job against NumPy master. 

```
        df = pd.DataFrame({"empty_col": Categorical([])})
>       result = df.describe()

pandas/tests/frame/test_analytics.py:594: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/core/generic.py:9956: in describe
    ldesc = [describe_1d(s) for _, s in data.iteritems()]
pandas/core/generic.py:9956: in <listcomp>
    ldesc = [describe_1d(s) for _, s in data.iteritems()]
pandas/core/generic.py:9939: in describe_1d
    return describe_categorical_1d(data)
pandas/core/generic.py:9900: in describe_categorical_1d
    objcounts = data.value_counts()
pandas/core/base.py:1318: in value_counts
    normalize=normalize, bins=bins, dropna=dropna)
pandas/core/algorithms.py:689: in value_counts
    result = Series(values)._values.value_counts(dropna=dropna)
pandas/core/arrays/categorical.py:1480: in value_counts
    count = bincount(obs, minlength=ncat or None)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (array([], dtype=int8),), kwargs = {'minlength': None}
relevant_args = (array([], dtype=int8), None)

>   ???
E   DeprecationWarning: 0 should be passed as minlength instead of None; this will error in future.

```

I would see if Categorical.value_counts can be updated to pass `count = bincount(obs, minlength=ncat or 0)`. If that doesn't work for older numpy's then you'll need a bit more compat code.

--- a/doc/source/whatsnew/v0.25.0.rst
+++ b/doc/source/whatsnew/v0.25.0.rst
@@ -184,6 +184,40 @@ are returned. (:issue:`21521`)
 
     df.groupby("a").ffill()
 
+``DataFrame`` describe on an empty categorical / object column will return top and freq
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+When calling :func:`describe`` on a :class:`DataFrame` with an empty categorical / object
+column, the 'top' and 'freq' columns were previously omitted, which was inconsistent with
+the output for non-empty columns. Now, the 'top' and 'freq' for the column will be shown
+as np.NaN in the case of an empty :class:`DataFrame` (:issue:`26397`)
+
+.. ipython:: python
+
+   df = pd.DataFrame({"empty_col": pd.Categorical([])})
+
+*Previous Behavior*:
+
+.. code-block:: python
+
+   In [3]: df.describe()
+   Out[3]:
+           empty_col
+   count           0
+   unique          0
+
+*New Behavior*:
+
+.. code-block:: python
+
+    In [4]: df.describe()
+    Out[4]:
+            empty_col
+    count         0.0
+    unique        0.0
+    top           NaN
+    freq          NaN
+
 
 .. _whatsnew_0250.api_breaking.deps:
 

--- a/pandas/core/arrays/categorical.py
+++ b/pandas/core/arrays/categorical.py
@@ -1477,7 +1477,7 @@ class Categorical(ExtensionArray, PandasObject):
 
         if dropna or clean:
             obs = code if clean else code[mask]
-            count = bincount(obs, minlength=ncat or None)
+            count = bincount(obs, minlength=ncat or 0)
         else:
             count = bincount(np.where(mask, code, ncat))
             ix = np.append(ix, -1)

--- a/pandas/tests/frame/test_analytics.py
+++ b/pandas/tests/frame/test_analytics.py
@@ -588,6 +588,8 @@ class TestDataFrameAnalytics:
         result = df3.describe()
         tm.assert_numpy_array_equal(result["cat"].values, result["s"].values)
 
+    def test_describe_empty_categorical_column(self):
+        # GH 26397
         # Ensure the index of an an empty categoric DataFrame column
         # also contains (count, unique, top, freq)
         df = pd.DataFrame({"empty_col": Categorical([])})

